DataFormats/TrackReco/interface/TrackBase.h

```diff
@@ -422,158 +449,36 @@ inline TrackBase::index TrackBase::covIndex(index i, index j)
-      break;
-    case lowPtTripletStep:
-      return "lowPtTripletStep";
-      break;
-    case pixelPairStep:
-      return "pixelPairStep";
-      break;
-    case detachedTripletStep:
-      return "detachedTripletStep";
-      break;
-    case mixedTripletStep:
-      return "mixedTripletStep";
-      break;
-    case pixelLessStep:
-      return "pixelLessStep";
-      break;
-    case tobTecStep:
-      return "tobTecStep";
-      break;
-    case jetCoreRegionalStep:
-      return "jetCoreRegionalStep";
-      break;
-    case conversionStep:
-      return "conversionStep";
-      break;
-    case muonSeededStepInOut:
-      return "muonSeededStepInOut";
-      break;
-    case muonSeededStepOutIn:
-      return "muonSeededStepOutIn";
-      break;
-    case outInEcalSeededConv:
-      return "outInEcalSeededConv";
-      break;
-    case inOutEcalSeededConv:
-      return "inOutEcalSeededConv";
-      break;
-    case nuclInter:
-      return "nuclInter";
-      break;
-    case standAloneMuon:
-      return "standAloneMuon";
-      break;
-    case globalMuon:
-      return "globalMuon";
-      break;
-    case cosmicStandAloneMuon:
-      return "cosmicStandAloneMuon";
-      break;
-    case cosmicGlobalMuon:
-      return "cosmicGlobalMuon";
-      break;
-    case iter1LargeD0:
-      return "iter1LargeD0";
-      break;
-    case iter2LargeD0:
-      return "iter2LargeD0";
-      break;
-    case iter3LargeD0:
-      return "iter3LargeD0";
-      break;
-    case iter4LargeD0:
-      return "iter4LargeD0";
-      break;
-    case iter5LargeD0:
-      return "iter5LargeD0";
-      break;
-    case bTagGhostTracks:
-      return "bTagGhostTracks";
-      break;
-    case gsf:
-      return "gsf";
-      break;
-    case hltPixel :
-      return "hltPixel";
-      break;
-    case hltIter0 :
-      return "hltIter0";
-      break;
-    case hltIter1 :
-      return "hltIter1";
-      break;
-    case hltIter2 :
-      return "hltIter2";
-      break;
-    case hltIter3 :
-      return "hltIter3";
-      break;
-    case hltIter4 :
-      return "hltIter4";
-      break;
-    case hltIterX :
-      return "hltIterX";
-      break;
-    default:
-      return "undefAlgorithm";
-      break;
-    }
+    return (TrackAlgorithm) (algorithm_);
+}
+inline TrackBase::TrackAlgorithm TrackBase::originalAlgo() const
+{
+    return (TrackAlgorithm) (originalAlgorithm_);
 }
 
+
+
+inline std::string TrackBase::algoName() const { return TrackBase::algoName(algo()); }
+
 inline bool TrackBase::quality(const TrackBase::TrackQuality q) const
 {
     switch (q) {
     case undefQuality:
-        return (quality_ == 0);
+        return quality_ == 0;
     case goodIterative:
-        return (((quality_ & (1 << TrackBase::confirmed))  >> TrackBase::confirmed) ||
-                ((quality_ & (1 << TrackBase::highPurity)) >> TrackBase::highPurity));
+        return (quality_ & (1 << TrackBase::highPurity)) >> TrackBase::highPurity;
```

Comment:
@VinInn 
remind me, please, why this change happened?
The return value logic for quality(goodIterative) has  changed: instead of an OR of bit2 (highPurity) and bit3 (confirmed), we now get only bit2.
quality(goodIterative) appears in a few places in CMSSW (git grep -i goodIterative).